Makefile

```
@@ -743,125 +705,51 @@ $(OCRD_EXECUTABLES:%=%-check):
-docker-mini%: DOCKER_MODULES := $(DOCKER_MODULES_MINI)
-# Medium-size selection: add Olena binarization and Calamari, add evaluation
-DOCKER_MODULES_MEDI := $(DOCKER_MODULES_MINI) cor-asv-ann dinglehopper docstruct format-converters nmalign ocrd_calamari ocrd_keraslm ocrd_olena ocrd_segment
-docker-medi%: DOCKER_MODULES := $(DOCKER_MODULES_MEDI)
-# Maximum-size selection: use all enabled modules
-docker-maxi%: DOCKER_MODULES := $(OCRD_MODULES)
-
-# DOCKER_BASE_IMAGE
-docker-minimum: DOCKER_BASE_IMAGE = ocrd/core:$(CORE_VERSION)
-docker-medium: DOCKER_BASE_IMAGE = $(DOCKER_TAG):minimum
-docker-maximum: DOCKER_BASE_IMAGE = $(DOCKER_TAG):medium
-# CUDA variants
-docker-minimum-cuda: DOCKER_BASE_IMAGE = ocrd/core-cuda:$(CORE_VERSION)
-docker-medium-cuda: DOCKER_BASE_IMAGE = $(DOCKER_TAG):minimum-cuda
-docker-maximum-cuda: DOCKER_BASE_IMAGE = $(DOCKER_TAG):medium-cuda
-# explicit interdependencies
-docker-medium: docker-minimum
-docker-maximum: docker-medium
-docker-medium-cuda: docker-minimum-cuda
-docker-maximum-cuda: docker-medium-cuda
-
-# Build rule for all selections
-# FIXME: $(DOCKER_MODULES) ref does not work at phase 1; workaround: all modules
-docker-%: Dockerfile modules
-	docker build \
-	--progress=plain \
-	--build-arg BASE_IMAGE=$(DOCKER_BASE_IMAGE) \
-	--build-arg VCS_REF=$$(git rev-parse --short HEAD) \
-	--build-arg BUILD_DATE=$$(date -u +"%Y-%m-%dT%H:%M:%SZ") \
-	--build-arg OCRD_MODULES="$(DOCKER_MODULES)" \
-	--build-arg PIP_OPTIONS="$(PIP_OPTIONS)" \
-	--build-arg PARALLEL="$(DOCKER_PARALLEL)" \
-	--build-arg PYTHON="$(PYTHON)" \
-	--network=host \
-	-t $(DOCKER_TAG):$* .
-
-docker: DOCKER_MODULES ?= $(OCRD_MODULES)
-docker: DOCKER_PARALLEL ?= -j1
-docker: docker-latest
 
+.PHONY: deps-ubuntu
+
+OCRD_NETWORK_CONFIG ?= run-network/ocrd-all-config.yaml
+
+.PHONY: network-setup network-start network-stop network-clean
+network-setup: init-vol-models
+network-setup: docker-compose.yml
+network-setup: .env
+network-setup: ocrd-processing-server-config.yaml
+
+docker-compose.yml: $(ACTIVATE_VENV) ocrd-all-images.yaml
+	. $(ACTIVATE_VENV) && python run-network/creator.py create-compose $(OCRD_NETWORK_CONFIG) $@
+.env: $(ACTIVATE_VENV)
+	. $(ACTIVATE_VENV) && python run-network/creator.py create-dotenv $(OCRD_NETWORK_CONFIG) $@
+	@# overrides
+	@echo RES_VOL=$(DOCKER_VOL_MODELS) >> $@
+	@echo USER_ID=`id -u` >> $@
+	@echo GROUP_ID=`id -g` >> $@
+#	. $(ACTIVATE_VENV) && python run-network/creator.py create-clients $(BIN) $(OCRD_NETWORK_CONFIG)
+ocrd-processing-server-config.yaml: $(ACTIVATE_VENV)
+	. $(ACTIVATE_VENV) && python run-network/creator.py create-psconfig $(OCRD_NETWORK_CONFIG) $@
+network-start:
+	# docker compose up -d
+	. $(ACTIVATE_VENV) && python run-network/creator.py start
+network-stop:
+	# docker compose down
+	. $(ACTIVATE_VENV) && python run-network/creator.py stop
+network-clean:
+	$(RM) -r $(VIRTUAL_ENV) .env docker-compose.yml ocrd-processing-server-config.yaml
+	docker volume rm $(DOCKER_VOL_MODELS)
```

```suggestion
	-docker volume rm $(DOCKER_VOL_MODELS)
```
sometimes cleanup fails for me when the models volume does not exist, which should not cause a failure

---

Indeed.